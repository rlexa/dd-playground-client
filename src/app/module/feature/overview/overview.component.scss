:host {
  display: flex;
  flex-flow: row wrap;
  place-content: flex-start;
  align-items: flex-start;
}
