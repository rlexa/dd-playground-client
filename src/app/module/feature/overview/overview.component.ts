import {CommonModule} from '@angular/common';
import {ChangeDetectionStrategy, Component} from '@angular/core';
import {MatCardModule} from '@angular/material/card';
import {MatListModule} from '@angular/material/list';

@Component({
  selector: 'app-overview',
  template: `<mat-card appearance="outlined">
    <mat-card-title>Empty?</mat-card-title>
    <mat-card-content>
      <mat-list>
        <h4 mat-header>Empty</h4>
      </mat-list>
    </mat-card-content>
  </mat-card>`,
  styleUrls: ['./overview.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  standalone: true,
  imports: [CommonModule, MatCardModule, MatListModule],
})
export class OverviewComponent {}
