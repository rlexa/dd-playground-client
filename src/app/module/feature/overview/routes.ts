import {Route} from '@angular/router';
import {RouteCurrent, RouteRoot, RouteWild} from 'src/app/routing';
import {NavigationBarItem, NavigationBarItemsData} from '../../widget/navigation-bar';
import {NavigationContentComponent} from '../../widget/navigation-content';

const routeNavs: Record<string, NavigationBarItem> = {
  [RouteCurrent]: {icon: 'current', route: RouteCurrent, label: 'Current'},
};

const data: NavigationBarItemsData = {
  navigationBarItems: Object.values(routeNavs),
};

export default [
  {
    path: RouteRoot,
    component: NavigationContentComponent,
    data,
    children: [
      {path: RouteCurrent, loadComponent: () => import('./overview.component').then((ii) => ii.OverviewComponent)},
      {path: RouteRoot, redirectTo: RouteCurrent, pathMatch: 'full'},
      {path: RouteWild, redirectTo: RouteCurrent},
    ],
  },
] as Route[];
