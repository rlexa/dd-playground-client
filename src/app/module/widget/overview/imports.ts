import { CommonModule } from '@angular/common';
import { MatButtonModule, MatCardModule, MatListModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { RoutedContentModule } from 'app/module/widget/routed-content';
import { FlexboxModule } from 'dd-flexbox';

export const imports = [CommonModule, RouterModule, MatButtonModule, MatCardModule, MatListModule, FlexboxModule, RoutedContentModule];
