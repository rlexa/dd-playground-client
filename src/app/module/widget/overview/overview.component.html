<mat-card>
  <mat-card-title>Empty?</mat-card-title>
  <mat-card-content>
    <mat-list>
      <h4 mat-header>Empty</h4>
    </mat-list>
  </mat-card-content>
</mat-card>
