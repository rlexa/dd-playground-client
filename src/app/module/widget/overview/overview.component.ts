import { ChangeDetectionStrategy, Component } from '@angular/core';
import { Router } from '@angular/router';
import { routeToAiApproxPolynom } from 'app/routing';

@Component({
  selector: 'app-overview',
  templateUrl: './overview.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class OverviewComponent {
  constructor(private readonly router: Router) { }
  gotoAiApproxPolynom = () => routeToAiApproxPolynom(this.router);
}
