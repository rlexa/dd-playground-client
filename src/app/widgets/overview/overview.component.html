<div ddFlexbox="vyt">
  <mat-card>
    <mat-card-title>Nothing to show yet.</mat-card-title>
    <mat-card-content>
    </mat-card-content>
  </mat-card>
</div>