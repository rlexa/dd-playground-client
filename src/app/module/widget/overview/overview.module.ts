import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { ROUTE_ROOT, ROUTE_WILDCARD } from 'app/routing';
import { imports } from './imports';
import { OverviewComponent } from './overview.component';

const ROUTING = [
  { path: ROUTE_ROOT, component: OverviewComponent },
  { path: ROUTE_WILDCARD, redirectTo: ROUTE_ROOT, pathMatch: 'full' },
];

@NgModule({ imports: [...imports, RouterModule.forChild(ROUTING)], exports: [OverviewComponent], declarations: [OverviewComponent] })
export class OverviewModule { }
