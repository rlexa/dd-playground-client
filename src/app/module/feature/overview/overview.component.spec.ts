import {ComponentFixture} from '@angular/core/testing';
import {MockBuilder, MockRender} from 'ng-mocks';
import {OverviewComponent} from './overview.component';

describe('OverviewComponent', () => {
  let fixture: ComponentFixture<OverviewComponent>;

  beforeEach(() => MockBuilder(OverviewComponent));

  beforeEach(() => {
    fixture = MockRender(OverviewComponent);
    fixture.detectChanges();
  });

  test('renders', () => expect(fixture).toMatchSnapshot());
});
