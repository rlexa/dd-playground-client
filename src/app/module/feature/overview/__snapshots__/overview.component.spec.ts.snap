// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewComponent renders 1`] = `
<mock-render>
  <app-overview>
    <mat-card
      appearance="outlined"
      ng-reflect-appearance="outlined"
    >
      <mat-card-title>
        Empty?
      </mat-card-title>
      <mat-card-content>
        <mat-list>
          <h4
            mat-header=""
          >
            Empty
          </h4>
        </mat-list>
      </mat-card-content>
    </mat-card>
  </app-overview>
</mock-render>
`;
